src/server/routes/get_tx.rs

```diff
@@ -1,16 +1,27 @@
+use std::sync::{atomic::Ordering, Arc};
+
 use actix_web::{web::Data, HttpResponse};
 
 use crate::{
     database::{models::Transaction, schema::transactions::dsl::*},
     server::error::ValidatorServerError,
+    state::{SharedValidatorState, ValidatorState},
     types::DbPool,
 };
 use diesel::prelude::*;
 
-pub async fn get_tx(
-    path: (String,),
+pub trait Config {
+    fn get_validator_state(&self) -> &SharedValidatorState;
+}
```

Comment:
We should probably put this trait into some common file, for example `context.rs` in where we collect some shared traits. Also, I think this trait should be called `StateAccess` or `ValidatorStateAccess` or something similar to have more descriptive name.